private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/maven/MavenResolver.java

```diff
@@ -0,0 +1,427 @@
+  }
+
+  @Override
+  public Graph<Coordinates> resolve(ResolutionRequest request) {
+    List<RemoteRepository> repos =
+        request.getRepositories().stream()
+            .map(remoteRepositoryFactory::createFor)
+            .collect(Collectors.toList());
+
+    List<Dependency> boms =
+        request.getBoms().stream().map(this::createBom).collect(Collectors.toList());
+    List<Dependency> dependencies =
+        request.getDependencies().stream().map(this::createDependency).collect(Collectors.toList());
+    Set<Exclusion> globalExclusions =
+        request.getGlobalExclusions().stream()
+            .map(this::createExclusion)
+            .collect(Collectors.toSet());
+
+    RepositorySystem system = createRepositorySystem();
+    ConsoleRepositoryListener consoleLog = new ConsoleRepositoryListener(listener);
+    ErrorReportingListener errors = new ErrorReportingListener();
+    CoordinateGatheringListener coordsListener = new CoordinateGatheringListener();
+    RepositorySystemSession session =
+        prepareSession(
+            system,
+            new DefaultDependencyManager(),
+            new CompoundListener(consoleLog, errors, coordsListener),
+            request.getLocalCache());
+
+    List<RemoteRepository> repositories = new ArrayList<>(repos.size());
+    repositories.add(createRemoteRepoFromLocalM2Cache(request.getLocalCache()));
+    repositories.addAll(repos);
+
+    List<Dependency> amendedDeps = addGlobalExclusions(globalExclusions, dependencies);
+    List<Dependency> amendedBoms = addGlobalExclusions(globalExclusions, boms);
+
+    consoleLog.setPhase("Resolving BOM artifacts");
+
+    List<Dependency> managedDependencies =
+        resolveArtifactsFromBoms(system, session, repositories, amendedBoms);
+
+    managedDependencies = overrideDependenciesWithUserChoices(managedDependencies, dependencies);
+
+    Artifact fakeRoot = new DefaultArtifact("com.example:bazel-dep-resolver:1.0.0");
+
+    Dependency dep = new Dependency(fakeRoot, JavaScopes.COMPILE);
+    DependencyCollectionContext depCollectionContext =
+        new DefaultDependencyCollectionContext(session, null, dep, managedDependencies);
+    DependencyManager derived =
+        new DefaultDependencyManager().deriveChildManager(depCollectionContext);
+    session =
+        prepareSession(
+            system,
+            derived,
+            new CompoundListener(consoleLog, errors, coordsListener),
+            request.getLocalCache());
+
+    consoleLog.setPhase("Gathering direct dependency coordinates");
+    List<DependencyNode> directDependencies =
+        resolveBaseDependencies(
+            system, session, repositories, fakeRoot, managedDependencies, amendedDeps);
+
+    List<Exception> exceptions = errors.getExceptions();
+    if (!exceptions.isEmpty()) {
+      Exception exception = exceptions.get(0);
+      if (exception instanceof ModelBuildingException) {
+        System.out.println("\n[WARNING] " + exception);
```

Comment:
I used System.out here but I don't think this is the right thing to do with what we're doing with the ConsoleListener output